lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -526,40 +526,111 @@ LogicalResult ConvertAtenOp<AtenArgmaxOp>::matchAndRewrite(
+public:
+  using OpConversionPattern<AtenOpT>::OpConversionPattern;
+  using OpAdaptor = typename AtenOpT::Adaptor;
 
-  Value self = adaptor.self();
-  auto selfTy = self.getType().template cast<RankedTensorType>();
+  // Each variant must implement corresponding parameter parsing options
+  virtual LogicalResult
+  generateSqueezedShape(AtenOpT op, RankedTensorType selfTy,
+                        ConversionPatternRewriter &rewriter,
+                        SmallVector<int64_t> &squeezedShape) const {
+    return rewriter.notifyMatchFailure(
+        op, "Unimplemented dim/dim-list parsing function");
+  }
 
-  if (!selfTy)
-    return op.emitError("Only ranked tensor types supported in TOSA argmax");
+  // Common rewriter for all squeeze ops, calls the specific implementation of
+  // generateSqueezedShape() needed for the op variant.
+  LogicalResult matchAndRewrite(AtenOpT op, OpAdaptor adaptor,
+                                ConversionPatternRewriter &rewriter) const {
+    Value self = adaptor.self();
+    auto selfTy = self.getType().template cast<RankedTensorType>();
+
+    if (!selfTy)
+      return op.emitError("Only ranked tensor types supported in TOSA argmax");
+
+    SmallVector<int64_t> newOutputShape;
+    if (failed(generateSqueezedShape(op, selfTy, rewriter, newOutputShape)))
+      return op.emitError("Squeeze could not compute new shape");
+
+    auto resultTy = OpConversionPattern<AtenOpT>::getTypeConverter()
+                        ->convertType(op.getResult().getType())
+                        .template cast<RankedTensorType>();
+    auto resultElemTy = resultTy.getElementType();
 
-  auto selfShape = selfTy.getShape();
+    auto newOutputTy = RankedTensorType::get(newOutputShape, resultElemTy);
 
-  SmallVector<int64_t> newOutputShape;
-  for (auto &dim : selfShape) {
-    if (dim != 1)
-      newOutputShape.push_back(dim);
+    auto reshapeOp = rewriter.create<tosa::ReshapeOp>(
+        op->getLoc(),
+        OpConversionPattern<AtenOpT>::getTypeConverter()->convertType(
+            newOutputTy),
+        self, rewriter.getI64ArrayAttr(newOutputShape));
+    rewriter.replaceOpWithNewOp<tensor::CastOp>(
+        op,
+        OpConversionPattern<AtenOpT>::getTypeConverter()->convertType(
+            newOutputTy),
+        reshapeOp);
+
+    return success();
   }
+};
 
-  auto resultTy = getTypeConverter()
-                      ->convertType(op.getResult().getType())
-                      .cast<RankedTensorType>();
-  auto resultElemTy = resultTy.getElementType();
+template <typename AtenOpT>
+class ConvertAtenSqueezeOneDimOp : public ConvertAtenSqueezeOp<AtenOpT> {
+  using ConvertAtenSqueezeOp<AtenOpT>::ConvertAtenSqueezeOp;
+  using OpAdaptor = typename AtenOpT::Adaptor;
+
+  LogicalResult
+  generateSqueezedShape(AtenOpT op, RankedTensorType selfTy,
+                        ConversionPatternRewriter &rewriter,
+                        SmallVector<int64_t> &squeezedShape) const {
```

Comment:
The two overrides of `generateSqueezedShape` are generating compiler warnings because they are not marked as `override`